cassandra/io/asyncorereactor.py

```diff
@@ -86,24 +88,22 @@ def maybe_start(self):
     def _run_loop(self):
         log.debug("Starting asyncore event loop")
         with self._loop_lock:
-            while True:
+            while not self._shutdown:
                 try:
-                    asyncore.loop(timeout=0.001, use_poll=True, count=1000)
+                    asyncore.loop(timeout=0.001, use_poll=True, count=100)
+                    self._timers.service_timeouts()
+                    if not asyncore.socket_map:
+                        time.sleep(0.005)
```

Comment:
I think it would be an alright idea to use a non-global map. I wasn't going to update that in this changeset since I really just need this loop to stay alive, global connections or not. I was staying with this to avoid having to manage a map at this level. If you think it's worth removing the dependency on socket_map in that module, I can look into it more.